Rename to sysl-go and fix issues with make test

cuminandpaprika opened the pull request:
- Add gitignore
- Rename references to `sysl-go-comms` to `sysl-go`
- Fix missing parameter in `makefile`
- Fix tests so that `make gen` and `make test` run correctly

Note: This PR will be **squash merged**

cuminandpaprika committed:
Add gitignore file

cuminandpaprika committed:
Add gitignore file

cuminandpaprika committed:
Rename reference sysl-go-comms to sysl-go

cuminandpaprika committed:
Rename reference sysl-go-comms to sysl-go

gkanz commented on .gitignore:
```diff
@@ -0,0 +1,6 @@
+.idea/*
+dist/*
+.DS_Store
+.vscode/
```
`.vscode/`, `.idea/*` and `.DS_Store` are not specific to the repo and should be in people's personal gitignores.

cuminandpaprika replied:
Good point, but is there any harm in leaving them here? I don't think we have any guides currently for setting up personal .gitignores

cuminandpaprika replied:
Happy to remove these once we have a reference guide for setting up personal .gitignores and circulate with the team/ wider eng org?

gkanz replied:
I don't think it's a good idea to pollute the repo `.gitignore`. Probably best to start as we intend to carry on.

camscale replied:
`man gitignore` : https://git-scm.com/docs/gitignore

cuminandpaprika replied:
Sure, fixed in https://github.com/anz-bank/sysl-go/pull/14/commits/2f01df7b6700f008fa2e8b9114dcdeaeb754ff47

gkanz commented on Makefile:
```diff
@@ -7,7 +7,7 @@ START=goFile
 
 define run-sysl
-sysl codegen --dep-path github.com/anz-bank/sysl-go-comms/$(TEST_DIR)/  --root . --root-transform . --transform $< --grammar $(GRAMMAR) --start $(START) --outdir $(OUT) --app-name $(APP) $(MODEL)
+sysl codegen --dep-path github.com/anz-bank/sysl-go/$(TEST_DIR)/  --root . --root-transform . --transform $< --grammar $(GRAMMAR) --start $(START) --outdir $(OUT) --app-name $(APP) $(MODEL)
```
Make is broken: `malformed module path "github.com/anz-bank/sysl-go-comms/codegen/tests//callback"`, not sure how it happened but it looks like `$(EXT_LIB_DIR)` fell off the end: `--dep-path github.com/anz-bank/sysl-go/$(TEST_DIR)/$(EXT_LIB_DIR)`.

cuminandpaprika replied:
@gkanz Yeah struggling to fix the make and test issues, I think that fixed a whole bunch of errors, thanks!

cuminandpaprika replied:
Fixed in https://github.com/anz-bank/sysl-go/pull/14/commits/a531f931ff08bc4ea2227e0f2e9907bb444abc48

gkanz commented on codegen/tests/simplegrpc/grpc_serviceinterface.go:
```diff
@@ -4,7 +4,7 @@ package simplegrpc
 import (
```
This file should be deleted.

cuminandpaprika replied:
Awesome, thanks will do!

cuminandpaprika replied:
Fixed in https://github.com/anz-bank/sysl-go/pull/14/commits/70cf7698bd11574944d9899c603cf44353ab29c7

gkanz commented on go.mod:
```diff
@@ -1,21 +1,18 @@
+module github.com/anz-bank/sysl-go
 
 require (
+	github.com/anz-bank/sysl-go-comms v0.0.0-20200313054710-9d2033a94714 // indirect
```
This looks like a problem.

cuminandpaprika replied:
Addressed in https://github.com/anz-bank/sysl-go/pull/14/commits/70cf7698bd11574944d9899c603cf44353ab29c7

cuminandpaprika replied:
Resolved!

gkanz reviewed (changes requested):
Please run make clean, gen, test and lint.

ChloePlanet commented:
The reason why rename it to `sysl-go`:

`sysl` to brand it as part of the sysl toolkit, `go` because it's specifically for Go codegen, not Java, Kotlin, etc. It’s more an integrations library, we combine two things here: the runtime libraries and the code generator.

We removed `server` because it's not ultimately just for server codegen(could be for a background job running under cron, or a cli client for downstreams) and `lib` because it's only partly a library and it's fairly uncommon to append lib to libraries anyway.

cuminandpaprika committed:
Add missing param to makefile

cuminandpaprika committed:
Add missing param to makefile

cuminandpaprika committed:
Delete unused file

cuminandpaprika committed:
Delete unused file

cuminandpaprika committed:
Rename simplegrpc to simpleproto

Fixes issue importing package

cuminandpaprika committed:
Rename simplegrpc to simpleproto

Fixes issue importing package

cuminandpaprika committed:
Run make clean and make gen

cuminandpaprika committed:
Run make clean and make gen

cuminandpaprika committed:
Skip failing tests due to missing cert files

Put back test.pb.go and test.proto. 
Skip failing tests, to be resolved in a separate issue
https://github.com/anz-bank/sysl-go/issues/17

Commit:
Merge branch 'chore/update-sysl-go-references' of https://github.com/cuminandpaprika/sysl-go-comms into chore/update-sysl-go-references

camscale commented on .gitignore:
```diff
@@ -0,0 +1,2 @@
+coverage*.out
+.sysl-tmp/*
```
I dont think this is needed. There is nothing that creates `.sysl-tmp` that I can see in the sysl repo or this one.

cuminandpaprika replied:
Good point, I've fixed this up in 8ada55f0ab451ef08f4edf2366310467fa084480 (will be doing a squash merge)

gkanz replied:
It came back into mine, but until someone can justify it I guess it's OK to remove.

Commit:
Tests will pass

camscale commented on codegen/testdata/simplegrpc/simplegrpc.proto:
```diff
@@ -1,6 +1,6 @@
 syntax  = "proto3";
 
-package simplegrpc;
+package simplepb;
```
Why is this changing? Does not appear to be part of sysl-go-comms -> sysl-go rename.

The commit title says it is renaming to simpleproto, but you did simplepb. Hard to review without any details, but this does not seem right.

gkanz replied:
I think the repo has lost a few changes from github.service.anz.

cuminandpaprika replied:
Good point, the PR seems to have evolved into renaming the sysl-go-comms as well as making sure that `make test` and `make gen` work correctly. 

I've updated the PR description to better reflect what it contains. 

The rename was required to get the module to import correctly for me, but it could be because I've got a differnt version of the *protoc* tool installed. I just tried reverting that commit and checking if the tests pass, and got an import error
```
go test -cover -tags codeanalysis ./...
go: finding github.com/anz-bank/sysl-go/codegen/tests/simplegrpc/simplegrpc latest
go: finding github.com/anz-bank/sysl-go/codegen/tests latest
go: finding github.com/anz-bank/sysl-go/codegen latest
go: finding github.com/anz-bank/sysl-go/codegen/tests/simplegrpc latest
# github.com/anz-bank/sysl-go/codegen/tests/simplegrpc
package github.com/anz-bank/sysl-go/codegen/tests/simplegrpc (test)
	imports github.com/anz-bank/sysl-go/codegen/tests/simplegrpc/simplegrpc: no matching versions for query "latest"
FAIL	github.com/anz-bank/sysl-go/codegen/tests/simplegrpc [setup failed]
?   	github.com/anz-bank/sysl-go/codegen/grammars	[no test files]
ok  	github.com/anz-bank/sysl-go/codegen/tests/dbendpoints	(cached)	coverage: 52.0% of statements
ok  	github.com/anz-bank/sysl-go/codegen/tests/multigrpc	0.024s	coverage: [no statements]
?   	github.com/anz-bank/sysl-go/codegen/tests/multigrpc/callback	[no test files]
?   	github.com/anz-bank/sysl-go/codegen/tests/multigrpc/cards	[no test files]
?   	github.com/anz-bank/sysl-go/codegen/tests/multigrpc/cardspb	[no test files]
?   	github.com/anz-bank/sysl-go/codegen/tests/multigrpc/wallet	[no test files]
ok  	github.com/anz-bank/sysl-go/codegen/tests/simple	0.026s	coverage: 37.9% of statements
?   	github.com/anz-bank/sysl-go/codegen/tests/simplegrpc/callback	[no test files]
?   	github.com/anz-bank/sysl-go/codegen/tests/simplegrpc/simple	[no test files]
?   	github.com/anz-bank/sysl-go/codegen/tests/simplegrpc/simplepb	[no test files]
?   	github.com/anz-bank/sysl-go/codegen/transforms	[no test files]
ok  	github.com/anz-bank/sysl-go/common	(cached)	coverage: 70.1% of statements
ok  	github.com/anz-bank/sysl-go/common/internal	(cached)	coverage: 87.2% of statements
ok  	github.com/anz-bank/sysl-go/config	(cached)	coverage: 70.8% of statements
ok  	github.com/anz-bank/sysl-go/convert	(cached)	coverage: 78.4% of statements
ok  	github.com/anz-bank/sysl-go/core	(cached)	coverage: 38.5% of statements
?   	github.com/anz-bank/sysl-go/core/build	[no test files]
?   	github.com/anz-bank/sysl-go/database	[no test files]
?   	github.com/anz-bank/sysl-go/handlerinitialiser	[no test files]
ok  	github.com/anz-bank/sysl-go/logging	(cached)	coverage: 58.3% of statements
?   	github.com/anz-bank/sysl-go/metrics	[no test files]
ok  	github.com/anz-bank/sysl-go/restlib	(cached)	coverage: 60.9% of statements
ok  	github.com/anz-bank/sysl-go/splunk	(cached)	coverage: 73.3% of statements
?   	github.com/anz-bank/sysl-go/status	[no test files]
ok  	github.com/anz-bank/sysl-go/validator	(cached)	coverage: 85.7% of statements
FAIL
make: *** [test] Error 1
```

camscale commented on core/testdata/proto/test.proto:
```diff
@@ -0,0 +1,15 @@
+
+message TestReply {
+  string field1 = 1;
+}
```
Missing eol at eof: https://gist.github.com/camh-/1bebfcff1b0f814e9b191edc60d5206b

cuminandpaprika replied:
Thanks, fixed in 46e858a

camscale commented on core/testdata/proto/test.proto:
```diff
@@ -0,0 +1,15 @@
+
+message TestReply {
+  string field1 = 1;
+}
```
Also, why is this file being added? It does not appear to be needed to rename the repo.

cuminandpaprika replied:
Yup agreed, it was added to ensure that `make test` would run successfully

gkanz reviewed (commented):
I have uploaded some changes to get the tests to pass. @ChloePlanet, could you please review the PR in its current state? Thanks.

camscale commented on codegen/tests/simplegrpc/simplegrpc_test.go:
```diff
@@ -5,8 +5,8 @@ import (
 
-	pb "github.com/anz-bank/sysl-go/codegen/tests/simplegrpc/grpc"
 	"github.com/anz-bank/sysl-go/codegen/tests/simplegrpc/simple"
+	pb "github.com/anz-bank/sysl-go/codegen/tests/simplegrpc/simplepb"
```
This commit should have been merged with the previous one, if this renaming of simplegrpc -> simplepb is really needed.

cuminandpaprika replied:
Will be doing a merge and squash for this PR :)

gkanz reviewed (approved):
LTGM

camscale reviewed (commented):
I've made some comments that I'm not sure how relevant they are. This would have been easier to review if it did just what the PR title said it would. If there are other breakages that were already present, they should have been fixed as separate PRs. Mixing together a bunch of unrelated changes makes it hard to review.

cuminandpaprika committed:
Remove unused gitignore reference

ChloePlanet commented on README.md:
```diff
@@ -6,7 +6,7 @@ Communication library used by SYSL-generated code written in Go.
 Go get the repository
 
-    go get github.com/anz-bank/sysl-go-comms
+    go get github.com/anz-bank/sysl-go
```
The first line of README needs change as well.
```diff
- # sysl-go-comms
+ # sysl-go
```

ChloePlanet replied:
There're several *.md still contain "sysl-go-comms". You can search "sysl-go-comms" through the whole project and replace them:)

cuminandpaprika replied:
Thanks, fixed in 682719e11aad09f76074240403f24c9eaa220025

ChloePlanet commented on codegen/tests/simple/simple_test.go:
```diff
@@ -154,7 +154,7 @@ func TestHandlerNotImplemented(t *testing.T) {
 	defer resp.Body.Close()
 	body, _ := ioutil.ReadAll(resp.Body)
-	require.JSONEq(t, `{"status":{"code":"1234", "description":"Internal Server Error"}}`, string(body))
+	require.JSONEq(t, `{"status":{"code":"9998", "description":"Internal Server Error"}}`, string(body))
```
Any reasons to change it back to 9998?

gkanz replied:
So the tests pass, again, I think there may have been some changes lost from github.service.anz.

cuminandpaprika replied:
I believe this was to get the tests to pass, but unclear about the details.

Could you please clarify @gkanz ?

ChloePlanet replied:
I've updated all status code to 1234 since Marcelo guesses it might be the internal status codes. If it isn't, 9998 is fine.

cuminandpaprika replied:
Ah, okay I think we should probably clarify this... Because it might mean we need to remove the proto files used by that test then.

cuminandpaprika replied:
Looks like the error codes are already here: https://github.com/anz-bank/sysl-go/blob/master/common/error_handler.go So the test is just verifying what's already there

cuminandpaprika replied:
@ChloePlanet Created an issue to refactor out the error codes in the future https://github.com/anz-bank/sysl-go/issues/19

Otherwise, could you re-review?

ChloePlanet commented on go.mod:
```diff
@@ -1,21 +1,18 @@
+module github.com/anz-bank/sysl-go
 
 require (
+	github.com/anz-bank/sysl-go-comms v0.0.0-20200313054710-9d2033a94714 // indirect
```
This import should be removed.

cuminandpaprika replied:
Yup thanks, removed in f62e203d25a8344b3efc3feffa11dc740d095ef1

cuminandpaprika committed:
Fix missed reference to sysl-go-comms

cuminandpaprika committed:
Add newline at end of proto

cuminandpaprika committed:
Run go mod tidy

cuminandpaprika committed:
Remove additional references to sysl-go-comms
